TAScheduler/views.py

```diff
@@ -135,6 +139,32 @@ def get(self, request):
             return render(request, "editcourse.html")
         else:
             return redirect("/../home/")
+
+class CreateLab(View):
+    def get(self, request):
+        # If the user does not have a valid name, I.E. if they try to manually enter /home in the search bar,
+        # they will fail the userAllowed test and be redirected back to the login page
+        # If the user is allowed then home is rendered like normal
+        if userAllowed(request, ["SUPERVISOR"]):
+            LabManagement.createLab(request, request.POST['labID'], request.POST['labName'],
+            request.POST['labHours'], request.POST['labLocation'], request.POST['labDays'], 
+            request.POST['labInstructor'], request.POST['labTA'])
+        else:
+            return redirect("/../home/")
```

Comment:
```suggestion
        if userAllowed(request, ["SUPERVISOR"]):
            return render(request, "createlab.html")
        else:
            return redirect("/../home/")
            
def post(self, request):
     LabManagement.createLab(request, request.POST['labID'], request.POST['labName'],
            request.POST['labHours'], request.POST['labLocation'], request.POST['labDays'], 
            request.POST['labInstructor'], request.POST['labTA'])
            return render(request, "createlab.html")
```
Form data needs to be processed in post whereas get handles the redirects and rendering of the page.